docs/entities/renderer.md

```diff
@@ -0,0 +1,584 @@
+public class MyRenderLayer extends RenderLayer<MyEntityRenderState, MyEntityModel> {
+    private final MyEntityModel model;
+    
+    public MyRenderLayer(MyEntityRenderer renderer, EntityModelSet entityModelSet) {
+        super(renderer);
+        this.model = new MyEntityModel(entityModelSet.bakeLayer(MyEntityModel.MY_LAYER));
+    }
+
+    @Override
+    public void render(PoseStack poseStack, MultiBufferSource bufferSource, int packedLight, MyEntityRenderState renderState, float yRot, float xRot) {
+        // ...
+    }
+}
+```
+
+```java
+public class MyEntityRenderer
+        extends LivingEntityRenderer<MyEntity, MyEntityRenderState>
+        implements RenderLayerParent<MyEntityRenderState, MyEntityModel> {
+    public MyEntityRenderer(EntityRendererProvider.Context context) {
+        super(context);
+        this.addLayer(new MyRenderLayer(this, context.getModelSet()));
+    }
+
+    public MyEntityRenderState createRenderState() {
+        return new MyEntityRenderState();
+    }
+
+    @Override
+    public void extractRenderState(MyEntity entity, MyEntityRenderState state, float partialTick) {
+        super.extractRenderState(entity, state, partialTick);
+        // ...
+    }
+
+    @Override
+    public void render(MyEntityRenderState state, PoseStack poseStack, MultiBufferSource bufferSource, int packedLight) {
+        super.render(state, entityYaw, partialTick, poseStack, bufferSource, packedLight);
+        // ...
+    }
+}
+```
+
+```java
+@SubscribeEvent
+public static void registerLayerDefinitions(EntityRenderersEvent.RegisterLayerDefinitions event) {
+    event.add(MyEntityModel.MY_LAYER, MyEntityModel::createBodyLayer);
+}
+
+@SubscribeEvent
+public static void registerEntityRenderers(EntityRenderersEvent.RegisterRenderers event) {
+    event.registerEntityRenderer(MY_ENTITY_TYPE.get(), MyEntityRenderer::new);
+}
+```
+
+## Modifying Existing Entity Renderers
+
+In some scenarios, it is desirable to add to an existing entity renderer, e.g. for rendering additional effects on an existing entity. Most of the time, this will affect living entities, i.e., entities with a `LivingEntityRenderer`. This enables us to add [render layers][renderlayer] to an entity like so:
+
+```java
+@SubscribeEvent
+public static void addLayers(EntityRenderersEvent.AddLayers event) {
+    // Add a layer to every single entity type.
+    for (EntityType<?> entityType : event.getEntityTypes()) {
+        // Get our renderer.
+        EntityRenderer<?, ?> renderer = event.getRenderer(entityType);
+        // Null-check the renderer. You could add more checks here,
+        // e.g., instanceof checks to only target specific renderer classes.
+        if (renderer != null) {
+            // Add the layer to the renderer. Like above, construct a new MyRenderLayer.
+            // The EntityModelSet can be retrieved from the event through #getEntityModels.
+            renderer.addLayer(new MyRenderLayer(renderer, event.getEntityModels()));
```

Comment:
This would not compile due to the generics.

Reply:
This still wouldn't compile. `MyRenderLayer` does not have the same generics as every entity renderer. Either create a new `RenderLayer` class for the example or only apply this to the appropriate entities.